tools/Az.Tools.Predictor/Az.Tools.Predictor/AzContext.cs

```diff
@@ -176,19 +228,16 @@ private Version GetAzVersion()
         /// <summary>
         /// Executes the PowerShell cmdlet in the current powershell session.
         /// </summary>
-        private static List<T> ExecuteScript<T>(string contents)
+        private List<T> ExecuteScript<T>(string contents)
         {
             List<T> output = new List<T>();
 
-            using (PowerShell powershell = PowerShell.Create(RunspaceMode.NewRunspace))
-            {
-                powershell.AddScript(contents);
-                Collection<T> result = powershell.Invoke<T>();
+            PowerShellRuntime.AddScript(contents);
```

Comment:
Since you are reusing the powershell object, make sure run `PowerShellRuntime.Commands.Clear()` before (or after) executing a script. Otherwise, the scripts executed by `AddScript` will accumulate.